controllers/cloudstackcluster_controller.go

```diff
@@ -69,52 +71,58 @@ func (r *CloudStackClusterReconciler) Reconcile(ctx context.Context, req ctrl.Re
 
 	// Get CloudStack cluster.
 	csCluster := &infrav1.CloudStackCluster{}
-	if retErr = r.Client.Get(ctx, req.NamespacedName, csCluster); retErr != nil {
-		if client.IgnoreNotFound(retErr) == nil {
+	if err := r.Client.Get(ctx, req.NamespacedName, csCluster); err != nil {
```

Comment:
Named vars can be tricky.

In our case, retErr is actually modified by the patch defer func if an error occurs during patching.

As such, we need to be careful not to re-declare retErr or the named return var functionality is lost.